llvm/lib/ExecutionEngine/Orc/LLJIT.cpp

```diff
@@ -949,8 +949,18 @@ Expected<std::unique_ptr<ObjectLayer>>
 LLJIT::createObjectLinkingLayer(LLJITBuilderState &S, ExecutionSession &ES) {
 
   // If the config state provided an ObjectLinkingLayer factory then use it.
-  if (S.CreateObjectLinkingLayer)
-    return S.CreateObjectLinkingLayer(ES);
+  if (S.CreateObjectLinkingLayer) {
+    auto Layer = S.CreateObjectLinkingLayer(ES);
+    if (Layer && S.JTMB->getTargetTriple().isOSBinFormatCOFF()) {
+      auto *RTDyldLayer =
+          dyn_cast_or_null<RTDyldObjectLinkingLayer>(Layer.get().get());
+      if (RTDyldLayer) {
+        RTDyldLayer->setOverrideObjectFlagsWithResponsibilityFlags(true);
+        RTDyldLayer->setAutoClaimResponsibilityForObjectSymbols(true);
+      }
+    }
+    return Layer;
+  }
```

Comment:
We can assume Layer is non-null on success, so this can just be:
```c++
  if (S.CreateObjectLinkingLayer) {
    auto Layer = S.CreateObjectLinkingLayer(ES);
    if (auto *RTDyldLayer = dyn_cast<RTDyldObjectLinkingLayer>(Layer->get())) {
      RTDyldLayer->setOverrideObjectFlagsWithResponsibilityFlags(true);
      RTDyldLayer->setAutoClaimResponsibilityForObjectSymbols(true);
    }
    return Layer;
  }
  ```

Reply:
Thanks for catching the `isOSBinFormatCOFF` check that I'd left out! :)